eradiate/scenes/biosphere.py

```diff
@@ -46,53 +47,379 @@ class BiosphereFactory(BaseFactory):
+
+            Unit-enabled field (default unit: cdu[length])
+
+        Parameter ``n_leaves`` (int):
+            Total number of leaves to generate. If ``size`` is set, it will override
+            this parameter. Default: 4000
+
+        Parameter ``hdo`` (float):
+            Mean horizontal distance between leaves. If ``size`` is set, it will
+            override this parameter. Default: 1.
+
+            Unit-enabled field (default unit: cdu[length])
+
+        Parameter ``hvr`` (float):
+            Ratio of mean horizontal leaf distance and vertical canopy extent.
+            If ``size`` is set, it will override this parameter. Default: 1.
+
+        Parameter ``seed`` (int):
+            Seed for the random number generator. Default: 1.
+
+        Parameter ``avoid_overlap`` (bool):
+            If ``True``, the scene element will attempt to place the leaves such
+            that they do not overlap. If a leaf cannot be placed without overlap
+            after 1e6 tries, it will raise a ``RuntimeError``. Default: False.
+
+            .. warning::
+
+                Depending on the canopy specification instantiation can take
+                a very long time if overlap avoidance is active!
+
+            .. note::
+
+               To emulate the behaviour of the raytran leaf cloud generator
+               simply try instantiating the leaf cloud several times with
+               different ``seed`` values and after a certain amount of
+               failures, run it without overlap avoidance.
+                """
+
+        # n_leaves or horizontal extent
+        if size[0] == 0 or size[1] == 0:
+            size[0] = size[1] = np.sqrt(
+                n_leaves * np.pi * leaf_radius ** 2 / lai)
+        else:
+            n_leaves = int(
+                np.floor(
+                (size[0] * size[1] * lai) / (np.pi * leaf_radius ** 2))
+            )
+
+        # hdo/hvr or vertical extent
+        if size[2] == 0:
+            size[2] = lai * hdo ** 3 / (
+                    np.pi * leaf_radius ** 2 * hvr)
+
+        leaf_positions, leaf_orientations = _create_leaf_cloud(
+            shape_type=shape_type,
+            cube_size=size,
+            sphere_radius=sphere_radius,
+            cylinder_radius=cylinder_radius,
+            cylinder_height=cylinder_height,
+            n_leaves=n_leaves,
+            leaf_radius=leaf_radius,
+            mu=mu,
+            nu=nu,
+            seed=seed,
+            avoid_overlap=avoid_overlap
+        )
+
+        return cls(
+            id=id,
+            leaf_positions=leaf_positions,
+            leaf_orientations=leaf_orientations,
+            leaf_reflectance=leaf_reflectance,
+            leaf_transmittance=leaf_transmittance,
+            leaf_radius=leaf_radius
+        )
 
-    @abstractmethod
     def shapes(self, ref=False):
-        pass
+        from eradiate.kernel.core import ScalarTransform4f, ScalarVector3f
+
+        kdu_length = kdu.get("length")
```

Comment:
```suggestion
        kernel_length = uck.get("length")
```
(`uck` stands for `unit_context_kernel`, defined in the `_units` private module and exposed in the  top-level module)